tests/test_set_config.py

```diff
@@ -399,3 +399,102 @@ def test_values_needed_f90nml(): #pylint: disable=unused-variable
+
+def test_cfg_to_yaml_conversion(): #pylint: disable=unused-variable
+    ''' Test that .a cfg file can be used to create a yaml object.'''
+    input_file = os.path.join(uwtools_file_base, pathlib.Path("fixtures/simple2_yaml.cfg"))
+
+    with tempfile.TemporaryDirectory(dir='.') as tmp_dir:
+
+        args = ['-i', input_file, '--dry_run', '--input_file_type', '.yaml']
+
+        outstring = io.StringIO()
+        with redirect_stdout(outstring):
+            set_config.create_config_obj(args)
+        result = outstring.getvalue()
+
+        expected = config.YAMLConfig(input_file)
+        expected_file = f'{tmp_dir}/test.yaml'
+        expected.dump_file(expected_file)
+
+        assert result.rstrip('\n') == str(expected)
+
+def test_output_file_conversion(): #pylint: disable=unused-variable
+    ''' Test that --output_input_type converts config object to desired object type'''
+    input_file = os.path.join(uwtools_file_base, pathlib.Path("fixtures/simple.nml"))
+
+    with tempfile.TemporaryDirectory(dir='.') as tmp_dir:
+
+        out_file = f'{tmp_dir}/test_ouput.cfg'
+        args = ['-i', input_file, '-o', out_file, '--output_file_type', '.nml']
+
+        set_config.create_config_obj(args)
+
+        expected = config.F90Config(input_file)
+        expected_file = f'{tmp_dir}/expected_nml.nml'
+        expected.dump_file(expected_file)
+
+        assert compare_files(expected_file, out_file)
+
+def test_config_file_conversion(): #pylint: disable=unused-variable
+    ''' Test that --config_input_type converts config object to desired object type'''
+    input_file = os.path.join(uwtools_file_base, pathlib.Path("fixtures/simple2.nml"))
+    config_file = os.path.join(uwtools_file_base, pathlib.Path("fixtures/simple2.ini"))
+
+    with tempfile.TemporaryDirectory(dir='.') as tmp_dir:
+
+        out_file = f'{tmp_dir}/test_config_conversion.nml'
+        args = ['-i', input_file, '-c', config_file, '-o', out_file, '--config_file_type', '.nml']
+
+        set_config.create_config_obj(args)
+
+        expected = config.F90Config(input_file)
+        config_file_obj = config.F90Config(config_file)
+        expected.update_values(config_file_obj)
+        expected_file = f'{tmp_dir}/expected_nml.nml'
+        expected.dump_file(expected_file)
+
+        assert compare_files(expected_file, out_file)
+
+def test_erroneous_conversion_flags(): #pylint: disable=unused-variable
+    ''' Test that error is thrown when conversion file types are not compatible'''
+
+    with tempfile.TemporaryDirectory(dir='.') as tmp_dir:
+
+        # test --input_file_type
+        input_file = os.path.join(uwtools_file_base, pathlib.Path("fixtures/simple2_nml.cfg"))
+        args = ['-i', input_file, '--input_file_type', ".pdf"]
+
+        outstring = io.StringIO()
+        with redirect_stdout(outstring):
+            set_config.create_config_obj(args)
+        result = outstring.getvalue()
+        outcome = "Set config failure: bad file type\n"
```

Comment:
There are some examples elsewhere to ensure that a specific exception is raised. That might be useful with the comment I left above.